pkg/pgmodel/migrate.go

```diff
@@ -457,14 +460,132 @@ func (t *Migrator) upgradeVersion(tx pgx.Tx, from, to semver.Version) error {
 	return nil
 }
 
-func installExtension(conn *pgxpool.Conn) {
-	_, extErr := conn.Exec(context.Background(), fmt.Sprintf(extensionInstall, extSchema))
-	if extErr != nil {
-		log.Warn("msg", "timescale_prometheus_extra extension not installed", "cause", extErr)
+func installExtension(conn *pgxpool.Conn) error {
+	availableVersions, err := getAvailableExtensionVersions(conn)
+	ExtensionIsInstalled = false
+	if err != nil {
+		return err
+	}
+	if len(availableVersions) == 0 {
+		log.Warn("msg", "timescale_prometheus_extra is not available, proceeding without extension")
+		return nil
+	}
+
+	newVersion, ok := getNewExtensionVersion(availableVersions)
+	if !ok {
+		log.Warn("msg", "timescale_prometheus_extra is not available at the right version, need version: %v, proceeding without extension", version.ExtVersionRangeString)
+		return nil
+	}
+
+	currentVersion, isInstalled, err := getInstalledExtensionVersion(conn)
+	if err != nil {
+		return fmt.Errorf("Could not get the installed extension version: %w", err)
+	}
+
+	if !isInstalled {
+		_, extErr := conn.Exec(context.Background(),
+			fmt.Sprintf("CREATE EXTENSION IF NOT EXISTS timescale_prometheus_extra WITH SCHEMA %s VERSION '%s'",
+				extSchema, getSqlVersion(newVersion)))
+		if extErr != nil {
+			return extErr
+		}
+		ExtensionIsInstalled = true
+		return extErr
+	}
+
+	comparator := currentVersion.Compare(newVersion)
+	if comparator > 0 {
+		//currentVersion greater than what we can handle, don't use the extension
+		log.Warn("msg", "timescale_prometheus_extra at a greater version than supported by the connector: %v > %v, proceeding without extension", currentVersion, newVersion)
 		ExtensionIsInstalled = false
+		return nil
+	} else if comparator == 0 {
+		//Nothing to do we are at the correct version
+		ExtensionIsInstalled = true
+		return nil
 	} else {
+		//Upgrade to the right version
+		_, err := conn.Exec(context.Background(),
+			fmt.Sprintf("ALTER EXTENSION timescale_prometheus_extra UPDATE TO '%s'",
+				getSqlVersion(newVersion)))
+		if err != nil {
+			return err
+		}
 		ExtensionIsInstalled = true
+		return nil
+	}
+}
+
+func getAvailableExtensionVersions(conn *pgxpool.Conn) (semver.Versions, error) {
```

Comment:
it bugs me a little that both this function, which reads from the DB, and `getNewExtensionVersion` below which does not, are both called `getFoo`, it feels like we should have a distinct prefix for stuff that reads. Maybe `fetch` as in `fetchAvailableExtensionVersions` and `fetchInstalledExtensionVersion`?